docs/usage/redux-logic-thunks.md

```diff
@@ -0,0 +1,443 @@
+  // fetchTodoByIdThunk is the "thunk function"
+  return async function fetchTodosThunk(dispatch, getState) {
+    const response = await client.get(`/fakeApi/todo/${todoId}`)
+    dispatch(todosLoaded(response.todos))
+  }
+}
+```
+
+Thunk functions and action creators can be written using either the `function` keyword or arrow functions - there's no meaningful difference here. The same `fetchTodoById` thunk could also be written using arrow functions, like this:
+
+```js title="Writing thunks using arrow functions"
+export const fetchTodoById = todoId => async dispatch => {
+  const response = await client.get(`/fakeApi/todo/${todoId}`)
+  dispatch(todosLoaded(response.todos))
+}
+```
+
+In either case, the thunk is dispatched by calling the action creator, in the same way as you'd dispatch any other Redux action:
+
+```js
+function TodoComponent({ todoId }) {
+  const dispatch = useDispatch()
+
+  const onFetchClicked = () => {
+    // Calls the thunk action creator, and passes the thunk function to dispatch
+    dispatch(fetchTodoById(todoId))
+  }
+}
+```
+
+### Why Use Thunks?
+
+Thunks allow us to write additional Redux-related logic separate from a UI layer. This logic can include side effects, such as async requests or generating random values, as well as logic that requires dispatching multiple actions or access to the Redux store state.
+
+Redux reducers [must not contain side effects](../tutorials/fundamentals/part-3-state-actions-reducers.md#rules-of-reducers), but real applications require logic that has side effects. Some of that may live inside components, but some may need to live outside the UI layer. Thunks (and other Redux middleware) give us a place to put those side effects.
+
+It's common to have logic directly in components, such as making an async request in a click handler or a `useEffect` hook and then processing the results. However, it's often necessary to move as much of that logic as possible outside the UI layer. This may be done to improve testability of the logic, to keep the UI layer as thin and "presentational" as possible, or to improve code reuse and sharing.
+
+In a sense, a thunk is a loophole where **you can write any code that needs to interact with the Redux store, ahead of time, without needing to know _which_ Redux store will be used**. This keeps the logic from being bound to any specific Redux store instance and keeps it reusable.
+
+Historically, another reason to use thunks was to help keep React components "unaware of Redux". The `connect` API allowed passing action creators and "binding" them to automatically dispatch actions when called. Since components typically did not have access to `dispatch` internally, passing thunks to `connect` made it possible for components to just call `this.props.doSomething()`, without needing to know if it was a callback from a parent, dispatching a plain Redux action, dispatching a thunk performing sync or async logic, or a mock function in a test.
```

Comment:
I think we've hit too many words and while this is interesting might be too much information, especially with connect not being recommended? Consider trimming?

Reply:
I can make it a "detailed explanation" expander, yeah